workflow/controller/operator.go

```diff
@@ -493,6 +493,14 @@ func (woc *wfOperationCtx) persistUpdates() {
 		woc.log.Warnf("Failed to dehydrate: %v", err)
 		woc.markWorkflowError(err, true)
 	}
+
```

Comment:
can I ask you to copy this check:

https://github.com/argoproj/argo/pull/3800/files#diff-fcb04129f1d32e69cca32631c6586587R485

but don't panic - instead log error - otherwise it is risky to back-port

Reply:
scrub this - I don't think we should make this PR more complex